packages/editor/src/components/header/index.js

```diff
@@ -168,7 +170,9 @@ function Header( {
 				/>
 
 				{ canBeZoomedOut && isWideViewport && (
-					<ZoomOutToggle disabled={ forceDisableBlockTools } />
+					<ZoomOutToggle
+						disabled={ forceDisableBlockTools || isDistractionFree }
+					/>
```

Comment:
It looks like other settings buttons in the header are hidden via CSS. I wonder if we should do the same here.

cc @draganescu

https://github.com/WordPress/gutenberg/blob/564f0699e0924b7114af55d51f1909de07a2e3c9/packages/editor/src/components/header/style.scss#L257-L261

Reply:
Hi @Mamaduka,

Based on the comment above, if I use CSS to hide the button, a potential issue arises: pressing cmd + shift + 0 will still trigger the zoom-out action via the shortcut, as the logic resides in the callback function within the ZoomOutToggle component.

```js
useShortcut( 'core/editor/zoom', () => {
        if ( isZoomOut ) {
	        resetZoomLevel();
        } else {
	        setZoomLevel( 'auto-scaled' );
        }
} );
```

To address this, I see two potential solutions:

1. Update the callback function inside useShortcut to include a check for isDistractionFree.
2. Avoid rendering the ZoomOutToggle component when isDistractionFree mode is enabled and not use CSS to hide.

I’d appreciate your input on which approach would be preferable.

Reply:
The `useShortcut` also accepts options argument, and we can disable the shortcut when `isDistractionFree` is true.

Example:

https://github.com/WordPress/gutenberg/blob/564f0699e0924b7114af55d51f1909de07a2e3c9/packages/editor/src/components/global-keyboard-shortcuts/index.js#L46-L56

Reply:
> I don't know why the other buttons are hidden with CSS

I don't know this either; it could be related to animations. Let's follow the current example from the core for hiding similar buttons; if the hiding method is changed in the future, we can do that in one go.